packages/react/menu/src/Menu.tsx

```diff
@@ -244,9 +268,42 @@ const MenuRootContent = React.forwardRef((props, forwardedRef) => {
   }, []);
 
   return (
-    <MenuContentImpl {...props} ref={composedRefs} onDismiss={() => context.onOpenChange(false)} />
+    <MenuContentImpl
+      {...props}
+      ref={composedRefs}
+      // we make sure we're not trapping once it's been closed
+      // (closed !== unmounted when animating out)
+      trapFocus={context.open}
+      // make sure to only disable pointer events when open
+      // this avoids blocking interactions while animating out
+      disableOutsidePointerEvents={context.open}
+      disableOutsideScroll
+      // When focus is trapped, a `focusout` event may still happen.
+      // We make sure we don't trigger our `onDismiss` in such case.
+      onFocusOutside={composeEventHandlers(
+        props.onFocusOutside,
+        (event) => event.preventDefault(),
+        { checkForDefaultPrevented: false }
+      )}
+      onDismiss={() => context.onOpenChange(false)}
+    />
   );
-}) as MenuRootContentPrimitive;
+}) as MenuRootContentTypePrimitive;
+
+const MenuRootContentNonModal = React.forwardRef((props, forwardedRef) => {
+  const context = useMenuContext(CONTENT_NAME);
+
+  return (
+    <MenuContentImpl
+      {...props}
+      ref={forwardedRef}
+      trapFocus={false}
```

Comment:
I need a refresher on this one. I can see this switches off trapping but doesn't enable looping. 

@benoitgrelard I thought we had said that if things were non-modal they would `loop` and not `trap` so that we didn't have to worry about moving focus to the element after the trigger?

Reply:
Ah I can see we do `loop` in popover. I'm guessing we don't here because it's a menu and that's all handled by `RovingFocusGroup` anyway?

Reply:
>  I'm guessing we don't here because it's a menu which tend not to loop

That was my understanding. Tab's currently [blocked on menus](https://github.com/radix-ui/primitives/blob/80dacf836f3be65941d0f3fa017991f75b1313a2/packages/react/menu/src/Menu.tsx#L544-L545) based on the decision we made when implementing the submenus. Do we think this should be different for non modal?

Reply:
No I think it's all correct because menus should really be treated as one tab stop only.

Reply:
Cool, if we're good with this it all looks fine to me :+1: